flagging_site/blueprints/api.py

```diff
@@ -0,0 +1,50 @@
+import pandas as pd
+from flask import Blueprint
+from flask import render_template
+from flagging_site.data.hobolink import get_hobolink_data
+from flagging_site.data.usgs import get_usgs_data
+from flagging_site.data.model import process_data
+from flagging_site.data.model import reach_2_model
+from flagging_site.data.model import reach_3_model
+from flagging_site.data.model import reach_4_model
+from flagging_site.data.model import reach_5_model
+from flask_restful import Resource, Api
+
+class ReachApi(Resource):
+
+    def model_api(self) -> dict:
+        """
+        Class method that retrieves data from hobolink and usgs and processes
+        data, then creates json-like dictionary structure for model output.
+
+        returns: json-like dictionary
+        """
+        df = get_data()
+
+        dfs = {
+            2: reach_2_model(df),
+            3: reach_3_model(df),
+            4: reach_4_model(df),
+            5: reach_5_model(df)
+        }
+
+        main = {}
+        models = {}
+
+        # adds metadata
+        main['version'] = '2020'
+        main['time_returned'] = str(pd.to_datetime('today'))
+
+        for reach, df in dfs.items():
+            add_to_dict(models, df, reach)
+
+        # adds models dict to main dict
+        main['models'] = models
+
+        return main
+
+    def get(self):
+        return self.model_api()
+
+
+api.add_resource(ReachApi, '/api/v1/model')
```

Comment:
This won't work as-is. `api` is not a defined object. Check Cameron's code again in `flagging.py` and see where he defines `api`, and also see _how_ he defines it. And then, with that information, combine that with my ask relating to the blueprint.

![image](https://user-images.githubusercontent.com/31971762/87211612-4a1f7c00-c2e8-11ea-952a-a1f61ec7f9eb.png)

The issue isn't just that `api` isn't defined, although that's the most important thing to fix. After you get the code 100% correct, there will be a very tiny change you need to make to this line to make it work. I trust you'll figure it out though ;)

Reply:
This works (although haven't made the tiny change to this particular line). I copied the way api is defined in flagging.py and mimicked for api.py.